cmd/crank/beta/validate/image.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2024 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validate
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/crane"
+	conregv1 "github.com/google/go-containerregistry/pkg/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+)
+
+// ImageFetcher defines an interface for fetching images
+type ImageFetcher interface {
+	FetchBaseLayer(image string) (*conregv1.Layer, error)
+}
+
+// Fetcher implements the ImageFetcher interface
+type Fetcher struct{}
+
+// FetchBaseLayer fetches the base layer of the image which contains the 'package.yaml' file
+func (f *Fetcher) FetchBaseLayer(image string) (*conregv1.Layer, error) {
+	cBytes, err := crane.Config(image)
+	if err != nil {
+		return nil, errors.Wrapf(err, "cannot get config")
+	}
+
+	cfg := &conregv1.ConfigFile{}
+	if err := yaml.Unmarshal(cBytes, cfg); err != nil {
+		return nil, errors.Wrapf(err, "cannot unmarshal image config")
+	}
+
+	var label string
+	ls := cfg.Config.Labels
+	for v, k := range ls {
+		if k == baseLayerLabel {
```

Comment:
this will indeed work, but actually what we care about is the annotation on the layers in the manifest, which is what we actually pull from the package manager [here](https://github.com/crossplane/crossplane/blob/d70cd5ebb0f3721893a7296762f91f274cd5414c/internal/controller/pkg/revision/imageback.go#L120), see [here](https://github.com/crossplane/crossplane/blob/b2b2a98338c2b81c8d5cc1787f29268057066ac7/internal/xpkg/layers.go#L102) for some comment on why we dd the labels also to the config. I think it would be more robust to rely on the same thing as the package manager, or at least add a TODO referencing this issue so that if we ever decide to change this behaviour, we know what might have happened